lib/deprecations_detector/collect.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module DeprecationsDetector
+  class Collect
+    def self.call(message = "", callstack = [], deprecation_horizon = nil, gem_name = nil)
+      DeprecationsDetector::Main.add_deprecation(message, caller)
+    end
+
+    def self.arity
+      4
+    end
```

Comment:
What is the significance of this method?

Reply:
This specifies the number of arguments required in rails 5.
it's required by the ActiveSupport Behavior

ref https://apidock.com/ruby/Proc/arity

Reply:
Cool